.github/workflows/lint-sample.yml

```diff
@@ -38,6 +38,7 @@ jobs:
       - name: Run linter
         uses: golangci/golangci-lint-action@v6
         with:
+          install-mode: goinstall
```

Comment:
I understand that allowing us to pass the CI and ensure that the lint works was required. 
Am I right? Would it work without this change?

Reply:
Yes.  Solved it done.

Reply:
Lets move forward but I create an issue to see if we can remove it before the next release or if we should leave it as it is now: https://github.com/kubernetes-sigs/kubebuilder/issues/4750

Reply:
This was not the right fix.

You are using golangci-lint v1.63, the first version of golangci-lint that supports go1.24 is golangci-lint v1.64.

The right fix is to update the golangci-lint version to a version that support go1.24

Reply:
Thanks @ldez 
Got it